src/main/java/nextstep/auth/LoginMemberAspect.java

```diff
@@ -0,0 +1,76 @@
+package nextstep.auth;
+
+import nextstep.auth.token.JwtTokenProvider;
+import nextstep.member.domain.Member;
+import nextstep.member.domain.MemberRepository;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Aspect
+@Component
+public class LoginMemberAspect {
+    public static final String AUTHORIZATION_BEARER = "bearer";
+    public static final String HEADER_AUTHORIZATION = "Authorization";
+    private final JwtTokenProvider jwtTokenProvider;
+    private final MemberRepository memberRepository;
+
+    public LoginMemberAspect(JwtTokenProvider jwtTokenProvider, MemberRepository memberRepository) {
+        this.jwtTokenProvider = jwtTokenProvider;
+        this.memberRepository = memberRepository;
+    }
+
+    @Around("@annotation(loginMember)")
+    public Object validateTokenAndInjectMember(ProceedingJoinPoint joinPoint, LoginMember loginMember) throws Throwable {
+        String username = validateTokenAndReturnUsername();
+        Object[] args = injectMember(joinPoint, username);
+
+        return joinPoint.proceed(args);
+    }
+
+    private String validateTokenAndReturnUsername() {
+        HttpServletRequest request =
+                ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+        String authorization = request.getHeader(HEADER_AUTHORIZATION);
+        validateAuthorization(authorization);
+
+        String token = authorization.split(" ")[1];
+        String username = jwtTokenProvider.getPrincipal(token);
+
+        return username;
+    }
+
+    private static void validateAuthorization(String authorization) {
+        if (!StringUtils.hasText(authorization)) {
+            throw new AuthenticationException();
+        }
+
+        if (authorization.split(" ").length != 2) {
+            throw new AuthenticationException();
+        }
+
+        if (!AUTHORIZATION_BEARER.equalsIgnoreCase(authorization.split(" ")[0])) {
+            throw new AuthenticationException();
+        }
+    }
+
+    private Object[] injectMember(ProceedingJoinPoint joinPoint, String username) {
+        Member member = memberRepository.findByEmail(username)
+                .orElseThrow(() -> new AuthenticationException());
+
+        Object[] args = joinPoint.getArgs();
+        for (int i = 0; i < args.length; i++) {
+            if (args[i] instanceof Member) {
+                args[i] = member;
+                break;
+            }
+        }
```

Comment:
```suggestion
        IntStream.range(0, args.length)
                .filter(i -> args[i] instanceof Member)
                .findFirst()
                .ifPresent(i -> args[i] = member);
```
2중 들여쓰기부분을 Stream을 활용해서 변환해볼 수도 있네요!
물론 지금도 충분합니다!

Reply:
오오 이런 방법이 있었네요! 감사합니다 😃